slidez-core/src/main/java/com/binarystudio/academy/slidez/app/user/UserController.java

```diff
@@ -12,50 +9,67 @@
-		if (token == null || token.isEmpty()) {
-			return new ResponseEntity<>("Invalid token", HttpStatus.BAD_REQUEST);
-		}
-
-		Optional<UserDetailsDto> userDetailsDto = this.userService.getDetailsByToken(token);
-		if (userDetailsDto.isEmpty()) {
-			return new ResponseEntity<>("Bad token.", HttpStatus.BAD_REQUEST);
-		}
-
-		return new ResponseEntity<>(userDetailsDto.get(), HttpStatus.OK);
-	}
-
-	@GetMapping("{id}")
-	public ResponseEntity<Object> getById(@PathVariable("id") UUID id) {
-		if (id == null) {
-			return new ResponseEntity<>("Invalid ID", HttpStatus.BAD_REQUEST);
-		}
-		Optional<User> userOptional = this.userService.getById(id);
-		if (userOptional.isEmpty()) {
-			return new ResponseEntity<>("User not found.", HttpStatus.NOT_FOUND);
-		}
-
-		User user = userOptional.get();
-		UserDetailsDto userDetailsDto = UserMapper.INSTANCE.mapUserToUserDetailsDto(user);
-		return new ResponseEntity<>(userDetailsDto, HttpStatus.OK);
-	}
-
-	@PutMapping("{id}")
-    public ResponseEntity<Object> updateUser(@PathVariable("id") UUID id,  User user) {
+    @Autowired
+    private UserService userService;
+
+    @GetMapping("")
+    public ResponseEntity<Object> findAll() {
+        return new ResponseEntity<>(userService.findAll(), HttpStatus.OK);
+    }
+
+    @GetMapping("userInfo")
+    public ResponseEntity<Object> getUserInfo(@RequestParam("token") String token) {
+        if (token == null || token.isEmpty()) {
+            return new ResponseEntity<>("Invalid token", HttpStatus.BAD_REQUEST);
+        }
+
+        Optional<UserDetailsDto> userDetailsDto = this.userService.getDetailsByToken(token);
+        if (userDetailsDto.isEmpty()) {
+            return new ResponseEntity<>("Bad token.", HttpStatus.BAD_REQUEST);
+        }
+
+        return new ResponseEntity<>(userDetailsDto.get(), HttpStatus.OK);
+    }
+
+    @GetMapping("{id}")
+    public ResponseEntity<Object> getById(@PathVariable("id") UUID id) {
         if (id == null) {
             return new ResponseEntity<>("Invalid ID", HttpStatus.BAD_REQUEST);
         }
+        Optional<User> userOptional = this.userService.getById(id);
+        if (userOptional.isEmpty()) {
+            return new ResponseEntity<>("User not found.", HttpStatus.NOT_FOUND);
+        }
+
+        User user = userOptional.get();
+        UserDetailsDto userDetailsDto = UserMapper.INSTANCE.mapUserToUserDetailsDto(user);
+        return new ResponseEntity<>(userDetailsDto, HttpStatus.OK);
+    }
+
+    @PostMapping("update")
+    public ResponseEntity<Object> updateUser(User user) {
         User userUpdated = userService.update(user);
         UserDetailsDto userDetailsDto = UserMapper.INSTANCE.mapUserToUserDetailsDto(userUpdated);
         return new ResponseEntity<>(userDetailsDto, HttpStatus.OK);
     }
 
-}
+    @PostMapping("")
+    public ResponseEntity<Object> updateUserByToken(@RequestParam("token") String token, User updatedUser) {
```

Comment:
Please, remove checking token in controller. We will have servlet filter to do that for all requests. If you want to test your API in Postman, just comment "applyRouteRestrictions" in SecurityConfig. And again, controller's response must be explisit